BuildingBlocks/test/BuildingBlocks.Application.Tests/Mediatr/QuotaEnforcerBehaviorTests.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Enmeshed.BuildingBlocks.Application.Abstractions.Exceptions;
+using Enmeshed.BuildingBlocks.Application.Attributes;
+using Enmeshed.BuildingBlocks.Application.MediatR;
+using Enmeshed.BuildingBlocks.Domain;
+using FluentAssertions;
+using MediatR;
+using Xunit;
+
+namespace Enmeshed.BuildingBlocks.Application.Tests.Mediatr;
+public class QuotaEnforcerBehaviorTests
+{
+    public QuotaEnforcerBehaviorTests() { }
+
+    [ApplyQuotasForMetrics("MetricKey")]
+    private class TestCommand : IRequest<IResponse> { }
+
+    private class IResponse { }
+
+    [Fact]
+    public async void Throws_QuotaExhaustedException_OnExhaustedQuota()
```

Comment:
I usually insert an underscore between each word (except for cohesive words like `QuotaExhaustedException`). And I would add even more details to the description: `Throws_QuotaExhaustedException_when_at_least_one_metric_is_exhausted`